advisor-ui/src/pages/Signup/Signup.jsx

```diff
@@ -60,8 +60,131 @@ export default function Signup() {
       locationState === "" ||
       address === ""
     ) {
+      return false;
+    }
+    return true;
+  };
+
+  const validateEmail = () => {
```

Comment:
You could probably accomplish all of this with a regex, something like `/^\S+@\S+\.\S+$/`.

Regexes to actually validate emails against the spec are pretty beefy but probably not worth worrying about. 

HTML forms/inputs might also have email as a type nowadays.